### DISCUSSION TITLE
Custom filtering

### DISCUSSION THREAD

**Discussion by koninos** [vmware/clarity#6449] [Q&A]
## Describe the bug

I need to create a custom filter that has a clear button to reset the filter value. The documentation is not very helpful on this since it does not provide a working example on custom filtering. Does anyone have a working example to help me with this issue?

**Comment by hippee-lee** [comment]
In the [custom filter docs](https://angular.clarity.design/documentation/datagrid/custom-filtering) take a look at the `ClrDatagridFilterInterface`. It can be used to create re-usable filters that can have any view template and functionality. 

The code for the custom filtering demos on the website is [here](https://github.com/vmware/clarity/tree/angular/projects/website/src/app/documentation/demos/datagrid/filtering). And the code for the [ColorFilter](https://github.com/vmware/clarity/blob/angular/projects/website/src/app/documentation/demos/datagrid/utils/color-filter.ts) you see on the last column is here.

**Reply by hippee-lee** [reply]
So for your question, you would use normal Angular bindings to coordinate with the filter value and the UI in such a way that it does what the app needs.
